roles/openstack-source/tasks/main.yml

```diff
@@ -7,14 +11,69 @@
   register: git_result
   until: git_result|success
   retries: 3
+
+- name: "python requirements for {{ project_name }}"
+  pip:
+    name: "{{ item.name }}"
+    version: "{{ item.version|default(omit) }}"
+    virtualenv: "{{ openstack_source.virtualenv_base }}/{{ project_name }}"
+    extra_args: "-i {{ openstack_source.pypi_mirror }}"
+  with_items: openstack_source.python_dependencies
+  when: openstack_source.python_dependencies is defined
```

Comment:
`openstack_source.python_dependencies` will always be defined, at least to an empty set, based on the `defaults.yml` content. I think this when statement is never going to not be true.